lib/SILOptimizer/SILCombiner/SILCombinerMiscVisitors.cpp

```diff
@@ -2237,3 +2237,18 @@ SILCombiner::visitCopyAddrInst(CopyAddrInst *CAI) {
       CAI->getLoc(), Src->getOperand(), Dst->getOperand(),
       CAI->isTakeOfSrc(), CAI->isInitializationOfDest());
 }
+
```

Comment:
Please implement this as instruction simplification in the SwiftCompilerSources (https://github.com/swiftlang/swift/tree/main/SwiftCompilerSources/Sources/Optimizer/InstructionSimplification).
We need to run it as part of the OnoneSimplificationPass in embedded swift. To do this, let `TypeValueInst` conform to `OnoneSimplifyable`. To also let it run in the SILCombine pass, let it conform to `SILCombineSimplifyable` and add `SWIFT_SILCOMBINE_PASS(TypeValueInst)` in `Passes.def`

Reply:
done!